File: Foundations & First Steps (2022)/February/gestor_notas/gestor_notas.py
# Gustavo Cocone Zavaleta | GitHub: @EgirlBuchona | © 2022
# Proyecto: Gestor de Notas en Consola en Python

notas = []

def agregar_nota():
    titulo = input("Título de la nota: ")
    contenido = input("Contenido de la nota: ")
    notas.append({"titulo": titulo, "contenido": contenido})
    print("Nota agregada exitosamente.\n")

def mostrar_notas():
    if not notas:
        print("No hay notas para mostrar.\n")
        return

    print("Listado de Notas:")
    for idx, nota in enumerate(notas, start=1):
        print(f"{idx}. {nota['titulo']}: {nota['contenido']}")
    print()

def eliminar_nota():
    mostrar_notas()
    try:
        indice = int(input("Ingrese el número de la nota a eliminar: ")) - 1
        if 0 <= indice < len(notas):
            notas.pop(indice)
            print("Nota eliminada exitosamente.\n")
        else:
            print("Número de nota inválido.\n")
    except ValueError:
        print("Por favor, ingrese un número válido.\n")

def main():
    while True:
        print("Gestor de Notas")
        print("1. Agregar Nota")
        print("2. Mostrar Notas")
        print("3. Eliminar Nota")
        print("4. Salir")
        opcion = input("Seleccione una opción: ")

        if opcion == "1":
            agregar_nota()
        elif opcion == "2":
            mostrar_notas()
        elif opcion == "3":
            eliminar_nota()
        elif opcion == "4":
            print("Saliendo del gestor de notas.")
            break
        else:
            print("Opción no válida. Intente nuevamente.\n")

if __name__ == "__main__":
    main()

File: (2022)/June/gestor_notas_java/GestorNotas.java
// Gustavo Cocone Zavaleta | GitHub: @EgirlBuchona | © 2022
// Proyecto: Gestor de Notas en Java

import java.util.ArrayList;
import java.util.Scanner;

public class GestorNotas {
    private static ArrayList<String> notas = new ArrayList<>();

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        boolean continuar = true;

        while (continuar) {
            System.out.println("\nGestor de Notas");
            System.out.println("1. Agregar nota");
            System.out.println("2. Eliminar nota");
            System.out.println("3. Buscar nota");
            System.out.println("4. Mostrar todas las notas");
            System.out.println("5. Salir");
            System.out.print("Seleccione una opción: ");
            int opcion = scanner.nextInt();
            scanner.nextLine();  // Consume la nueva línea

            switch (opcion) {
                case 1:
                    System.out.print("Ingrese la nota a agregar: ");
                    String nota = scanner.nextLine();
                    agregarNota(nota);
                    break;
                case 2:
                    System.out.print("Ingrese el índice de la nota a eliminar: ");
                    int indiceEliminar = scanner.nextInt();
                    eliminarNota(indiceEliminar);
                    break;
                case 3:
                    System.out.print("Ingrese una palabra clave para buscar: ");
                    String palabraClave = scanner.nextLine();
                    buscarNota(palabraClave);
                    break;
                case 4:
                    mostrarNotas();
                    break;
                case 5:
                    continuar = false;
                    System.out.println("Saliendo del gestor de notas.");
                    break;
                default:
                    System.out.println("Opción no válida.");
                    break;
            }
        }
        scanner.close();
    }

    public static void agregarNota(String nota) {
        notas.add(nota);
        System.out.println("Nota agregada exitosamente.");
    }

    public static void eliminarNota(int indice) {
        if (indice >= 0 && indice < notas.size()) {
            notas.remove(indice);
            System.out.println("Nota eliminada exitosamente.");
        } else {
            System.out.println("Índice no válido.");
        }
    }

    public static void buscarNota(String palabraClave) {
        System.out.println("Resultados de búsqueda:");
        for (int i = 0; i < notas.size(); i++) {
            if (notas.get(i).contains(palabraClave)) {
                System.out.println("Índice: " + i + " - Nota: " + notas.get(i));
            }
        }
    }

    public static void mostrarNotas() {
        System.out.println("Lista de notas:");
        for (int i = 0; i < notas.size(); i++) {
            System.out.println("Índice: " + i + " - Nota: " + notas.get(i));
        }
    }
}
